src/java/htsjdk/samtools/SAMUtils.java

```diff
@@ -945,14 +945,14 @@ public static int getMateUnclippedEnd(final SAMRecord rec) {
             } else {
                 if (getMateCigarString(rec) != null) {
                     ret = new ArrayList<SAMValidationError>();
-                    if (rec.getMateUnmappedFlag()) {
+                    if (!rec.getReadPairedFlag()) {
```

Comment:
We don't know why the if statement above was false; it's either because the read is not paired or the mate is unmapped.
If the statement failed because the read is not paired, then when it gets to this if statement here (in the original version) it will call rec.getMateUnmappedFlag() - but that will throw an exception because the code there has this check:
        requireReadPaired();
So, we need to switch the order here and have the if make sure that the read is paired.
Make sense?